GenerateAssemblyList failing with could not find file ...dependencylist.txt

To repro on an Ubuntu machine fetch master, init-tools, then build one test with /t:BuildAndTest.

Expect: run test

Actual: build fails with
```
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018: The "GenerateAssemblyList" task failed unexpectedly. [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018: System.IO.FileNotFoundException: Could not find file '/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/bin/TestDependencies/Microsoft.CSharp.Tests-AnyOS.AnyCPU.Debug.netcoreapp1.0.dependencylist.txt'. [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018: File name: '/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/bin/TestDependencies/Microsoft.CSharp.Tests-AnyOS.AnyCPU.Debug.netcoreapp1.0.dependencylist.txt' [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter) [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at Interop.CheckIo[TSafeHandle](TSafeHandle handle, String path, Boolean isDirectory, Func`2 errorRewriter) [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at System.IO.UnixFileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent) [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at System.IO.UnixFileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent) [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.GenerateAssemblyList.Execute() [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/Tools/tests.targets(213,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_debug_prtest/src/Microsoft.CSharp/tests/Microsoft.CSharp.Tests.csproj]
```

--- a/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets
+++ b/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets
@@ -182,19 +182,6 @@
       </IncludedFileForRunnerScript>
     </ItemGroup>
     
-    <!-- Introduce workaround for Dependency list to prevent copying local assemblies now that test csproj references pkgproj
-         This will remove all bin directory assemblies when TestWithLocalLibraries is false and local files are found.
-         This was added to allow the version of buildtools (that the buildtools repo depends on to be built) to be updated.  -->
-    <ItemGroup>
-      <IncludedFileForRunnerScriptRemove Include="@(IncludedFileForRunnerScript)"
-                                   Condition="$([System.String]::Copy('%(IncludedFileForRunnerScript.PackageRelativePath)').StartsWith('$(BinDir.Substring(0, $(BinDir.IndexOf('/'))))'))" >
-      </IncludedFileForRunnerScriptRemove>
-    </ItemGroup>
-    
-    <ItemGroup>
-      <IncludedFileForRunnerScript Condition="'$(TestWithLocalLibraries)'!='true'" Remove="@(IncludedFileForRunnerScriptRemove)" />
-    </ItemGroup>
-    
     <MakeDir Directories="$(OutputFolderForTestDependencies)" />
     <PropertyGroup>
       <_TestDependencyListRoot>$(AssemblyName)-$(OSGroup).$(Platform).$(ConfigurationGroup)</_TestDependencyListRoot>

